### ISSUE TITLE
Add custom HTTP headers into response error

### ISSUE THREAD

**Issue by NicoPennec** [nuxt/nuxt.js#7213]
### What problem does this feature solve?

When I throw an error in Nuxt.js, I can only customize `statusCode` and `message` properties.
But sometimes I need to add [HTTP headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers) to the response.

So I would like to set a custom header list for the error, eg. `Retry-After: 120` (see [google recommendation](https://webmasters.googleblog.com/2011/01/how-to-deal-with-planned-site-downtime.html))

### What does the proposed changes look like?

Set the [`Retry-After` header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After) on error.

eg. in a page.vue:
```vue
<script>
export default {
  asyncData () {
    try {
      // ...
    } catch (error) {
      if (error.statusCode === 429) {
        error.headers = { 'Retry-After': 120 }
      }
      throw error
    }
  }
}
</script>
```

or in a nuxt module:

```js
throw {
  statusCode: 503,
  message: 'Your request timed out. Please retry in a few minutes',
  headers: { 'Retry-After': 120 }
}
```

<div align="right"><sub><em>This feature request is available on <a href="https://cmty.app/nuxt">Nuxt</a> community (<a href="https://cmty.app/nuxt/nuxt.js/issues/c10518">#c10518</a>)</em></sub></div>

**Comment by pi0** [comment]
This feature is implemented by #7214 and waiting for release

**Comment by NicoPennec** [comment]
Feature shipped in [Nuxt v2.13.0](https://fr.nuxtjs.org/guide/release-notes/#v2.13.0) 🚀 

```js
// Basic example of a custom HTTP header

const error = new Error('My awesome error message')
error.headers = { 'Retry-After': 120 }
throw error
```
